io/disk/parallel_dd.py

```diff
@@ -0,0 +1,231 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM.
+# Author: Pooja B Surya <pooja@linux.vnet.ibm.com>
+# Author: Rajashree Rajendran<rajashr7@linux.vnet.ibm.com>
+
+# Based on code by Martin J. Bligh <mbligh@google.com>
+#   Copyright: 2007 Google
+#   https://github.com/autotest/autotest-client-tests/tree/master/parallel_dd
+
+"""
+Measures the performance of writing and reading multiple streams of files onto
+the files system.
+"""
+
+import os
+import time
+import sys
+import logging
+import json
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import partition as partition_lib
+
+
+class ParallelDd(Test):
+    """
+    Avocado test for parallel_dd.
+    """
+    def setUp(self):
+        """
+        :params disk: The disk on which the operations are to be performed.
+        :params fsys: A L{utils.partition} instance.
+        :params megabytes: The amount of data to read/write.
+        :params blocks: The blocksize in bytes to use.
+        :params streams: Number of streams. Defaults to 2.
+        :params blocks_per_file: The number of blocks per file.
+        :params fstype: The file system type of the disk.
+        :params seq_read: Perform sequential operations. Defaults to true.
+        :params dd_woptions: dd write options.
+        :params dd_roptions: dd read options.
+        :params fs_dd_woptions: dd write in streams.
+        :params fs_dd_roptions: dd read in streams.
+        """
+
+        self.disk = self.params.get('disk')
+        if not self.disk:
+            self.error('Test requires disk parameter,Please check README')
+        self.fsys = partition_lib.Partition(self.disk, mountpoint=self.srcdir)
+        self.megabytes = self.params.get('megabytes', default=100)
+        self.blocks = self.params.get('blocks', default=None)
+        self.streams = self.params.get('streams', default=2)
+        self.blocks_per_file = self.params.get(
+            'blocks_per_file', default=None)
+        self.fstype = self.params.get('fstype', default=None)
+        self.seq_read = self.params.get('seq_read', default=True)
+        self.dd_woptions = self.params.get('dd_woptions', default='')
+        self.dd_roptions = self.params.get('dd_roptions', default='')
+        self.fs_dd_woptions = self.params.get('fs_dd_woptions', default='')
+        self.fs_dd_roptions = self.params.get('fs_dd_roptions', default='')
+        if not self.blocks:
+            self.blocks = self.megabytes * 256
+
+        if not self.blocks_per_file:
+            self.blocks_per_file = self.blocks / self.streams
+
+        root_fs_device_cmd = "df | egrep /$ | awk '{print $1}'"
+        root_fs_device = process.system_output(root_fs_device_cmd, shell=True)
```

Comment:
can `root_fs_device = process.system_output('df | egrep /$ | awk '{print $1}', shell=True)` just want to avoid extra variable ?